src/generator/exercises/forth.cr

```diff
@@ -0,0 +1,56 @@
+require "./exercise_generator"
+require "./exercise_test_case"
+
+class ForthGenerator < ExerciseGenerator
+  def exercise_name
+    "forth"
+  end
+
+  def test_cases
+    test_cases = [] of JSON::Any
+    JSON.parse(data).each do |k, v|
+      v.each do |a, b|
+        test_cases.concat(b) if a.as_s == "cases"
+      end
+    end
+    test_cases.map do |test_case|
+      ForthTestCase.new(test_case)
+    end
+  end
+end
+
+class ForthTestCase < ExerciseTestCase
+  private getter input : JSON::Any | String
+  private getter description : JSON::Any
+  private getter expected : JSON::Any
+
+  def initialize(test_case)
+    @input = test_case["input"].as_a.join
+    @description = test_case["description"]
+    @expected = fix_empty_array(test_case["expected"])
+  end
+
+  def workload
+    if !(expected == nil)
+      "Forth.evaluate(#{input.inspect}).should eq(#{expected})"
+    else
+      <<-WL
```

Comment:
Just preference, and a habit I carried over from Ruby where you can do some neat things with heredocs, not sure if that carries over to Crystal